scheduler/utilization_based_host_allocator.go

```diff
@@ -54,15 +75,71 @@ func UtilizationBasedHostAllocator(ctx context.Context, hostAllocatorData HostAl
 	return newHostsNeeded, nil
 }
 
+// groupByTaskGroup takes a list of hosts and tasks and returns them grouped by task group
+func groupByTaskGroup(runningHosts []host.Host, taskQueue []model.TaskQueueItem) map[string]taskGroupData {
+	tgs := map[string]taskGroupData{}
+	for _, h := range runningHosts {
+		groupString := ""
+		if h.RunningTask != "" && h.RunningTaskGroup != "" {
+			groupString = makeTaskGroupString(h.RunningTaskGroup, h.LastBuildVariant, h.LastProject, h.LastVersion)
```

Comment:
I think these should be the RunningTask* equivalents of these Last* fields.

Reply:
fixed